scripts/pull_request_blaster_outer/new_ruby_api_service.rb

```diff
@@ -0,0 +1,158 @@
+  before_script:
+  - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
+  - chmod +x ./cc-test-reporter
+  - "./cc-test-reporter before-build"
+  - bundle exec rake db:create db:migrate
+  after_script:
+  - "./cc-test-reporter after-build --exit-code $TRAVIS_TEST_RESULT"
+  deploy:
+    provider: script
+    script: bash .travis/deploy.sh
+    on:
+      branch: master
+EOF
+
+puts "Creating .travis/deploy.sh file..."
+FileUtils.mkdir_p(".travis")
+File.write(".travis/deploy.sh", <<~EOF, :perm => 0755)
+  #!/bin/bash
+
+  TAG="$DEPLOY_SERVER/$DEPLOY_PROJECT/$DEPLOY_IMAGE:latest"
+
+  echo $DEPLOY_TOKEN | docker login -u $DEPLOY_USER --password-stdin $DEPLOY_SERVER
+  docker build -t $TAG .
+  docker push $TAG
+EOF
+
+puts "Modifying README.md for Travis Build Status..."
+readme = File.read("README.md")
+unless readme.include?("[![Build Status]")
+  parts = readme.split("\n")
+  parts.insert(2, "[![Build Status](https://travis-ci.org/#{ENV["GITHUB_REPO"]}.svg)](https://travis-ci.org/#{ENV["GITHUB_REPO"]})")
+  parts.insert(3, "") unless parts[3].start_with?("[![")
+  File.write("README.md", parts.join("\n"))
+end
+
+#
+# CodeClimate
+#
+
+require "rest-client"
+require "json"
+headers = {:accept => "application/vnd.api+json", :authorization => "Token token=#{ENV["CODECLIMATE_API_TOKEN"]}"}
+
+puts "Enabling CodeClimate..."
+resp = RestClient.get("https://api.codeclimate.com/v1/repos?github_slug=#{ENV["GITHUB_REPO"]}", headers)
+if JSON.parse(resp)["data"].empty?
+  payload = {"data" => {"type" => "repos", "attributes" => {"url" => "https://github.com/#{ENV["GITHUB_REPO"]}"}}}.to_json
+  resp = RestClient.post("https://api.codeclimate.com/v1/github/repos", payload, headers)
+end
+resp = JSON.parse(resp)
+
+badge_token = resp["data"][0]["attributes"]["badge_token"]
+test_reporter_id = resp["data"][0]["attributes"]["test_reporter_id"]
+
+puts "Modifying README.md for CodeClimate Maintainability..."
+readme = File.read("README.md")
+unless readme.include?("[![Maintainability]")
+  parts = readme.split("\n")
+  parts.insert(3, "[![Maintainability](https://api.codeclimate.com/v1/badges/#{badge_token}/maintainability)](https://codeclimate.com/github/#{ENV["GITHUB_REPO"]}/maintainability)")
+  File.write("README.md", parts.join("\n"))
+end
+
+puts "Modifying README.md for CodeClimate Test Coverage..."
+readme = File.read("README.md")
+unless readme.include?("[![Test Coverage]")
+  parts = readme.split("\n")
+  parts.insert(4, "[![Test Coverage](https://api.codeclimate.com/v1/badges/#{badge_token}/test_coverage)](https://codeclimate.com/github/#{ENV["GITHUB_REPO"]}/test_coverage)")
+  File.write("README.md", parts.join("\n"))
+end
+
+puts "Enabling Travis / CodeClimate integration"
+`travis env set CC_TEST_REPORTER_ID '#{test_reporter_id}' --org`
+
+puts "Removing .codeclimate.yml if present..."
+`git rm .codeclimate.yml` if File.exist?(".codeclimate.yml")
```

Comment:
@jrafanie I am removing any .codeclimate.yml files due to the new way that code climate works, since the file we have presently is legacy.  We may have to come up with something new.

Reply:
I'm not sure what you mean.  How does it work now?